build-static.sh

```diff
@@ -187,14 +187,16 @@ export CGO_CPPFLAGS="${CGO_CFLAGS}"
 if [ "${os}" = "mac" ]; then
 	export CGO_LDFLAGS="-framework CoreFoundation -framework SystemConfiguration"
 elif [ "${os}" = "linux" ] && [ -z "${DEBUG_SYMBOLS}" ]; then
-	CGO_LDFLAGS="-Wl,-O1 -pie"
+	CGO_LDFLAGS="-Wl,-O1 -pie -Wl,--allow-multiple-definition -Wl,--export-dynamic"
 fi
 
 CGO_LDFLAGS="${CGO_LDFLAGS} ${PWD}/buildroot/lib/libbrotlicommon.a ${PWD}/buildroot/lib/libbrotlienc.a ${PWD}/buildroot/lib/libbrotlidec.a ${PWD}/buildroot/lib/libwatcher-c.a $(${spcCommand} spc-config "${PHP_EXTENSIONS}" --with-libs="${PHP_EXTENSION_LIBS}" --libs)"
 if [ "${os}" = "linux" ]; then
 	if echo "${PHP_EXTENSIONS}" | grep -qE "\b(intl|imagick|grpc|v8js|protobuf|mongodb|tbb)\b"; then
 		CGO_LDFLAGS="${CGO_LDFLAGS} -lstdc++"
 	fi
+	CGO_LDFLAGS=$(echo "$CGO_LDFLAGS" | sed 's|-lphp|-Wl,--whole-archive -lphp -Wl,--no-whole-archive|g')
+	ar d ${PWD}/buildroot/lib/libphp.a $(ar t ${PWD}/buildroot/lib/libphp.a | grep '\.a$')
```

Comment:
include all symbols from -lphp (including "unused" ones that will later be used to load extensions)
remove all lib*.a files from libphp.a, because with the --whole-archive the linker isn't able to unpack them to pull in the object files later.

We don't need to unpack and rebuild the other .a object files, because they are all separately listed after -lphp and we don't need all of their symbols.